File: ios/moneypaypro/DevModeModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DevModeModule, NSObject)

RCT_EXTERN_METHOD(isDevModeEnabled:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/moneypaypro/DevModeModule.swift
import Foundation
import React

@objc(DevModeModule)
class DevModeModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func isDevModeEnabled(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    // On iOS, "Developer Mode" toggle (iOS 16+) isn't easily accessible from third party app.
    // For now, we'll return false or you could check if the app is a debug build.
    #if DEBUG
      resolve(true)
    #else
      resolve(false)
    #endif
  }
}
